### PR TITLE
Hooks for frequently made operations

### PR DESCRIPTION
**Pull request by joel-jeremy**
Add hooks for frequently made operations in the codebase:

- `useAccounts`
- `useAccount`
- `useBudgetedAccounts`
- `useOffBudgetAccounts`
- `useClosedAccounts`
- `useFailedAccounts`
- `usePayees`
- `usePayee`
- `useLocalPrefs`
- `useLocalPref`
- `useGlobalPrefs`
- `useGlobalPref`
- `useDateFormat`

### CODE REVIEW

File: packages/desktop-client/src/hooks/useAccount.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { useMemo } from 'react';
+
+import { useAccounts } from './useAccounts';
+
+export function useAccount(id: string) {
+  const accounts = useAccounts();
+  return useMemo(() => accounts.find(a => a.id === id), [id]);
```
Comment (MatissJanis) [inline packages/desktop-client/src/hooks/useAccount.ts:7]: **🔨 warning:** ‏current implementation won't re-compute if account list changes.
```suggestion
  return useMemo(() => accounts.find(a => a.id === id), [id, accounts]);
```

File: packages/desktop-client/src/components/modals/LoadBackup.jsx [hunk 2]
```diff
@@ -55,10 +56,12 @@ export function LoadBackup({
...
 }) {
   const [backups, setBackups] = useState([]);
+  const prefsBudgetId = useLocalPref('id');
+  const budgetIdToLoad = budgetId || prefsBudgetId;
 
   useEffect(() => {
-    send('backups-get', { id: budgetId }).then(setBackups);
-  }, [budgetId]);
+    send('backups-get', { id: budgetIdToLoad }).then(setBackups);
+  }, [budgetIdToLoad]);
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/modals/LoadBackup.jsx:64]: **❓ question:** ‏why do we need this functional change?

File: packages/desktop-client/src/hooks/useLocalPref.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { useSelector } from 'react-redux';
+
+import { type LocalPrefs } from 'loot-core/types/prefs';
+
+export function useLocalPref<K extends keyof LocalPrefs>(
+  prefName: K,
+): LocalPrefs[K] {
+  return useSelector(state => state.prefs.local?.[prefName]);
```
Comment (MatissJanis) [inline packages/desktop-client/src/hooks/useLocalPref.ts:8]: **🥜 nitpick:** ‏my wishlist item (but don't feel pressured to actually implement this) - for these hooks to have both setter & getter and a default value.

```
const [myPref, setMyPref] = useLocalPref('budget.collapsed', true);

// Usage:
console.log('my pref:', myPref);
<Button onClick={(someNewValue) => setMyPref(someNewValue)}>Click me</Button>
```

**Review by MatissJanis — CHANGES_REQUESTED** [review]
Very exciting! This clean-up was long overdue. Thanks for undertaking it.

File: packages/desktop-client/src/components/modals/LoadBackup.jsx (hunk 2 above)
Comment (joel-jeremy) [reply packages/desktop-client/src/components/modals/LoadBackup.jsx:64]: Originally, the budget id was being retrieved by the root `Modals` component and passed to this component. I just moved that retrieval logic to this component.

File: packages/desktop-client/src/hooks/useLocalPref.ts (hunk 3 above)
Comment (joel-jeremy) [reply packages/desktop-client/src/hooks/useLocalPref.ts]: This is a good idea! Implemented this :)

**Comment by joel-jeremy** [comment]
What changed in the VRT screenshot is the default visible column on smaller screens. This defaults to the Budgeted column instead of the Spent. I've also removed some props that were passed in several layers deep in favor of the new hooks.

**Review by MatissJanis — APPROVED** [review]
Much cleaner! Great job!

**Review by MatissJanis — APPROVED** [review]

**Pull request merged**